lib/providers/products_provider.dart

```diff
@@ -43,8 +43,25 @@ class ProductsProvider with ChangeNotifier {
 
   Product findById(String id) => _items.firstWhere((prod) => prod.id == id);
 
+  // addProduct is called when the user adds a new product in the shop
+  // newProduct is used here because product is a final variable and can't be changed
   void addProduct(Product product) {
-    _items.add(product);
+    final newProduct = Product(
+      title: product.title,
+      description: product.description,
+      price: product.price,
+      imageUrl: product.imageUrl,
+      id: DateTime.now().toString(),
+    );
+    _items.add(newProduct);
     notifyListeners();
   }
+
+  void updateProduct(String id, Product newProduct) {
+    final prodIndex = _items.indexWhere((prod) => prod.id == id);
+    if (prodIndex >= 0) {
+      _items[prodIndex] = newProduct;
+      notifyListeners();
+    }
```

Comment:
is checking of the index greater than 0 really needed here?

Reply:
yes, this condition must have a boolean. it throws an error whenever I remove the condition.

Reply:
alright, let's keep it for now.